client.go

```diff
@@ -144,15 +144,16 @@ func (c *Client) handleErrorResp(resp *http.Response) error {
 	var errRes ErrorResponse
 	err := json.NewDecoder(resp.Body).Decode(&errRes)
 	if err != nil || errRes.Error == nil {
-		reqErr := RequestError{
+		reqErr := &RequestError{
 			HTTPStatusCode: resp.StatusCode,
 			Err:            err,
 		}
 		if errRes.Error != nil {
-			reqErr.Err = errRes.Error
+			reqErr.Err = fmt.Errorf(errRes.Error.Message)
```

Comment:
Wouldn't we lose the ability to use `errors.Is` and `errors.As` here? https://pkg.go.dev/errors

Reply:
you are right. I have updated the code and the error message to cover this case and still match the expected error result illustrated in the existing tests